relay-server/src/services/store.rs

```diff
@@ -269,6 +284,26 @@ impl StoreService {
             }
         }
 
+        if let (Some(replay_event), Some(replay_recording)) = (replay_event, replay_recording) {
```

Comment:
This is not exhaustive.  Consider the states `(Some(replay_event), None)` and `(None, Some(replay_recording))`.  In these cases we would need to produce the individual events.

Reply:
We should also have test coverage for these branches.